src/Subscriptions/Subscriber.php

```diff
@@ -19,79 +23,98 @@ class Subscriber
     /**
      * @var mixed
      */
-    public $context;
+    public $root;
 
     /**
      * @var array
      */
     public $args;
 
     /**
-     * @var string
+     * @var mixed
      */
-    public $operationName;
+    public $context;
 
     /**
-     * @var string
+     * @var \GraphQL\Language\AST\DocumentNode
      */
-    public $queryString;
+    public $query;
 
     /**
-     * @var mixed
+     * @var string
      */
-    public $root;
+    public $operationName;
 
     /**
-     * Create new subscription instance.
-     *
-     * @param  array  $args
+     * @param  mixed[]  $args
      * @param  \Nuwave\Lighthouse\Support\Contracts\GraphQLContext  $context
-     * @param  \GraphQL\Type\Definition\ResolveInfo  $info
-     * @param  string  $queryString
+     * @param  \GraphQL\Type\Definition\ResolveInfo  $resolveInfo
+     * @return void
      *
-     * @return static
+     * @throws \Nuwave\Lighthouse\Exceptions\SubscriptionException
      */
-    public static function initialize(
+    public function __construct(
         array $args,
         GraphQLContext $context,
-        ResolveInfo $info,
-        string $queryString
-    ): self {
-        if ($info->operation->name === null) {
-            throw new SubscriptionException('An operation name must be present on a subscription request.');
+        ResolveInfo $resolveInfo
+    ) {
+        $operationName = $resolveInfo->operation->name;
+        if (! $operationName) {
+            throw new SubscriptionException(
+                self::MISSING_OPERATION_NAME
+            );
         }
+        $this->operationName = $operationName->value;
 
-        $instance = new static;
+        $this->channel = self::uniqueChannelName();
+        $this->args = $args;
+        $this->context = $context;
 
-        $instance->channel = $instance->uniqueChannelName();
-        $instance->context = $context;
-        $instance->args = $args;
-        $instance->operationName = $info->operation->name->value;
-        $instance->queryString = $queryString;
-
-        return $instance;
+        $documentNode = new DocumentNode([]);
+        $documentNode->definitions = $resolveInfo->fragments;
+        $documentNode->definitions[] = $resolveInfo->operation;
+        $this->query = $documentNode;
     }
 
     /**
-     * Unserialize subscription.
+     * Unserialize subscription from a JSON string.
      *
      * @param  string  $subscription
-     *
-     * @return static
+     * @return $this
      */
-    public static function unserialize(string $subscription): self
+    public function unserialize($subscription): self
```

Comment:
What can this be other than string? Either add that in `@param` or typehint it to `string` again.

Reply:
The interface does not allow that.